### ISSUE TITLE
google_compute_instance: google_compute_instance.junk: Use metadata_startup_script instead of a startup-script key in "metadata"

### ISSUE THREAD

**Issue by mikemcrill** [hashicorp/terraform#9999]
### Terraform Version
`Terraform v0.7.9`

### Affected Resource(s)
Please list the resources as a list, for example:
- google_compute_instance

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "google_compute_instance" "junk" {
  name           = "inf-junk"
  machine_type   = "n1-standard-4"
  can_ip_forward = false
  zone           = "us-central1-b"

  tags = ["server"]

  disk {
    image = "centos-7"
  }

  network_interface {
    subnetwork = "us-central1"
    access_config {
      // Ephemeral IP
    }
  }

  metadata {
    startup-script = "${data.template_file.startup-script.rendered}"
  }
}
```

### Expected Behavior
This should work per the docs so we don't have to recreate instances when we change the startup script:
metadata_startup_script - (Optional) An alternative to using the startup-script metadata key, except this one forces the instance to be recreated (thus re-running the script) if it is changed. This replaces the **startup-script metadata key** on the created instance and thus the two mechanisms are not allowed to be used simultaneously.

### Actual Behavior
Forces you to use `metadata_startup_script` which forces a rebuild of the instance.

### References
https://github.com/hashicorp/terraform/blob/master/builtin/providers/google/resource_compute_instance.go#L1005

**Comment by cblecker** [comment]
This could potentially be fixed by the changes @evandbrown proposed in #8407.

**Comment by paddyforan** [comment]
This should be addressed by #10537. :)
